### ISSUE TITLE
"curl -h" truncated

### ISSUE THREAD

**Issue by gvanem** [curl/curl#5547]
I've just noticed that a simple `curl -h | most` (or `more`, `less` or `> file`) stops listing the 
help after the line `--tls13-ciphers <ciphersuite list> TLS 1.3 c` (line 211).
Not the full 233 lines it should be.

I've always thought that the CRT will register an `atexit()` handler to flush standard buffers when `exit()` is called.  But the MSVC runtime seems not.

Adding a `fflush(stdout);` at the end of `` tool_help()` fixes it.

I'm on Win-10 using MSVC 2019 (v19.26.28806 for x86). Compiled using `-MD`.

**Comment by jay** [comment]
That's weird, how exactly did you build it?

**Comment by gvanem** [comment]
Weird, yes. I built using a home-brew Makefile. It's been working for years.
But some recent change somewhere has broken this particular issue.

**Comment by jay** [comment]
We're going to need more than that. Can you reproduce using any of the regular build methods?

**Comment by gvanem** [comment]
I've tried to use the official `winbuild/Makefile.vc` to reproduce, but failed (probably since my setup involves a lot more options and external libraries). But I see that `fflush()` is called on most streams, but not for `stdout`  when only e.g. `curl -h | less` is used.

I looked around and found [this](https://wiki.sei.cmu.edu/confluence/display/c/FIO23-C.+Do+not+exit+with+unflushed+data+in+stdout+or+stderr) advice on why it's good practice to
`fflush(stdout)`. A `fclose(stdout)` at the end of `tool_help.c` also prevent a truncated output.

And when building with `cl -MT`, the auto-flushing of `stdout` is more irregular. It seems there is a 12kByte buffer inside the `libcmt.lib`. This makes sense since the above `--tls13-ciphers <ciphersuite list> TLS 1.3 c` is roughly at the 12 kByte mark.
